src/common/netif/netif_haiku.c

```diff
@@ -0,0 +1,64 @@
+#include "netif.h"
+#include "common/io/io.h"
+
+#include <arpa/inet.h>
+#include <net/if.h>
+#include <net/route.h>
+#include <sys/socket.h>
+#include <sys/sockio.h>
+#include <stdio.h>
+
+// losely based on Haiku's src/bin/network/route/route.cpp
+
+bool ffNetifGetDefaultRouteImpl(char iface[IF_NAMESIZE + 1], uint32_t* ifIndex)
+{
+    // TODO: AF_INET6
+    FF_AUTO_CLOSE_FD int pfRoute = socket(AF_INET, SOCK_RAW, AF_INET);
+    if (pfRoute < 0)
+        return false;
+
+    struct ifconf config;
+    config.ifc_len = sizeof(config.ifc_value);
+    if (ioctl(pfRoute, SIOCGRTSIZE, &config, sizeof(struct ifconf)) < 0)
+        return false;
+
+    int size = (size_t)config.ifc_value;
+    if (size == 0)
+        return false;
+
+    void *buffer = malloc(size);
```

Comment:
FF_AUTO_FREE

Reply:
Yeah I suspected there was something like that to use, thanks.